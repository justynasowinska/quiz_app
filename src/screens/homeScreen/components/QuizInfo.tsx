import * as React from 'react';
import { Text, StyleSheet } from 'react-native';

import { THEME_COLORS } from '../../../config/colors';
import { QuestionType } from '../../../redux/types';

interface PropsType {
    questionsNumber: number;
    questionsType: QuestionType;
}

const getQuestionTypeInfo = (type: QuestionType) => {
    switch (type) {
        case 'boolean':
            return 'True or False';
        case 'multiple':
            return 'Multiple';
    }
};

export const QuizInfo = (props: PropsType) => {
    const { questionsNumber, questionsType } = props;
    return (
        <Text
            style={styles.info}
        >
            {`You will be presented with ${questionsNumber} ${getQuestionTypeInfo(questionsType)} questions`}.
        </Text>
    );
};

const styles = StyleSheet.create({
    info: {
        fontSize: 18,
        color: THEME_COLORS.appTextColor,
        textAlign: 'center'
    },
});