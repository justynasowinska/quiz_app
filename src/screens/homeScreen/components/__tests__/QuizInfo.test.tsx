import * as React from 'react';
import { shallow } from 'enzyme';

import { QuizInfo } from '../QuizInfo';

describe('Home Screen Quiz Info Component', () => {
    it('matches to the snapshots', () => {
        const component = shallow(
            <QuizInfo
                questionsNumber={4}
                questionsType="boolean"
            />
        );

        expect(component).toMatchSnapshot();

        component.setProps({ questionsNumber: 54, questionsType: 'multiple' });
        expect(component).toMatchSnapshot();
    });
});