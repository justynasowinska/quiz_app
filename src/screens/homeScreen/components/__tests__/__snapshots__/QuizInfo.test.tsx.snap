// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Quiz Info Component matches to the snapshots 1`] = `
<Text
  style={
    Object {
      "color": "#474D54",
      "fontSize": 18,
      "textAlign": "center",
    }
  }
>
  You will be presented with 4 True or False questions
  .
</Text>
`;

exports[`Home Screen Quiz Info Component matches to the snapshots 2`] = `
<Text
  style={
    Object {
      "color": "#474D54",
      "fontSize": 18,
      "textAlign": "center",
    }
  }
>
  You will be presented with 54 Multiple questions
  .
</Text>
`;
